klippa/tests/integration_test.rs

```diff
@@ -0,0 +1,538 @@
+        let output_dir = output_temp_dir.path();
+        for font in &self.fonts {
+            //TODO: add support for profiles later
+            //for profile in &self.profiles {
+            //}
+            for subset in &self.subsets {
+                //TODO: add support for instances/iup_options
+                self.run_one_test(font, subset, output_dir);
+            }
+        }
+    }
+
+    pub fn gen_expected_output(&self) {
+        //let output_dir = PathBuf::from("./qxliu_test");
+        //fs::create_dir("./qxliu_test").unwrap();
+        let output_temp_dir = TempDir::new_in(".", "klippa_test").unwrap();
+        let output_dir = output_temp_dir.path();
+        for font in &self.fonts {
+            //TODO: add support for profiles later
+            //for profile in &self.profiles {
+            //}
+            for subset in &self.subsets {
+                //TODO: add support for instances/iup_options
+                self.gen_expected_output_for_one_test(font, subset, output_dir);
+            }
+        }
+        let expected_dir = Path::new(TEST_DATA_DIR)
+            .join("expected")
+            .join(&self.expected_dir);
+        fs::rename(output_dir, expected_dir).unwrap();
+    }
+
+    fn run_one_test(&self, font: &str, subset: &str, output_dir: &Path) {
+        //TODO: re-enable subset="*" once populate_unicodes_to_retain supports *
+        if subset == "*" {
+            return;
+        }
+        let subset_font_name = gen_subset_font_name(font, subset);
+        let output_file = output_dir.join(&subset_font_name);
+        gen_subset_font_file(font, subset, &output_file);
+
+        let expected_file = Path::new(TEST_DATA_DIR)
+            .join("expected")
+            .join(&self.expected_dir)
+            .join(&subset_font_name);
+        compare_with_expected(output_dir, &output_file, &expected_file);
+    }
+
+    fn gen_expected_output_for_one_test(&self, font: &str, subset: &str, output_dir: &Path) {
+        //TODO: re-enable subset="*" once populate_unicodes_to_retain supports *
+        if subset == "*" {
+            return;
+        }
+
+        let subset_font_name = gen_subset_font_name(font, subset);
+        let output_file = output_dir.join(&subset_font_name);
+        gen_subset_font_file(font, subset, &output_file);
+
+        assert_has_ttx_exec();
+        let mut expected_file_name = String::from(&subset_font_name);
+        expected_file_name.push_str(".expected");
+        let expected_file = output_dir.join(expected_file_name);
+
+        let mut unicodes_option = String::from("--unicodes=");
+        unicodes_option.push_str(subset);
+
+        let mut output_option = String::from("--output-file=");
+        output_option.push_str(expected_file.to_str().unwrap());
+
+        let mut org_font_file = String::new();
+        org_font_file.push_str(TEST_DATA_DIR);
+        org_font_file.push_str("/fonts/");
+        org_font_file.push_str(font);
```

Comment:
I would just work with paths here? like,

```rust
let org_font_file = Path::new(TEST_DATA_DIR).join(fonts).join(font);
```

which isn't perfectly efficient (both `join` calls allocate) but is nice and concise at least.